src/libs/actions/templates.ts

```diff
@@ -6,7 +6,7 @@ import { inngest } from '@/inngest/client';
 import { creditTransactions, generated_templates, templateGallery, templates, users } from '@/models/Schema';
 import contentGenerator from '@/service/contentGenerator';
 import { generatePDFBuffer } from '@/service/generatePDFBuffer';
-import type { FetchTemplateResponse, FetchTemplatesRequest, GeneratedTemplates, GeneratePdfRequest, JsonObject, JsonValue, PaginatedResponse, TemplateType, UpdatePreviewURLParams, UpdatePreviewURLResult, UsageMetric, UsageMetricRequest } from '@/types/Template';
+import type { FetchTemplateResponse, FetchTemplatesRequest, GeneratedTemplates, GeneratePdfRequest, JsonObject, JsonValue, PaginatedResponse, UpdatePreviewURLParams, UpdatePreviewURLResult, UsageMetric, UsageMetricRequest } from '@/types/Template';
```

Comment:
lets create TemplateRequest.ts and TemplateResponse.ts and move relevant types there